### PR TITLE
Updated to new swagger spec, using vendoring, using latest version for go-swagger, and added get pipeline by ID

### PR DESCRIPTION
**Pull request by nicolaifsf**
using latest swagger spec, updated pipelines, vendoring using glide, go-swagger master branch is now used. Windows binary is built and released.

### CODE REVIEW

**Comment by nicolaifsf** [comment]
Look at PR 16 first

File: swagger.json [hunk 1]
```diff
@@ -1 +1 @@
...
...int":{"unique":true},"items":{"type":"string"}}},"type":"object"},"Model 4":{"type":"object","$ref":"#/definitions/Execution information"},"Build Webhook":{"properties":{"meta":{"type":"object","description":"Key=>Value information from the build itself"},"status":{"type":"string","description":"Current status of the build","x-example":"SUCCESS","default":"QUEUED","enum":["SUCCESS","FAILURE","QUEUED","ABORTED","RUNNING"]}},"required":["status"],"type":"object"},"Update Build":{"properties":{"status":{"type":"string","description":"Current status of the build","x-example":"SUCCESS","default":"QUEUED","enum":["SUCCESS","FAILURE","QUEUED","ABORTED","RUNNING"]}},"required":["status"],"type":"object"},"Update Job":{"properties":{"state":{"type":"string","description":"Current state of the Job","x-example":"ENABLED","default":"ENABLED","enum":["ENABLED","DISABLED"]}},"type":"object"},"Update Step Metadata":{"properties":{"code":{"type":"integer","description":"Exit code"},"startTime":{"type":"string","description":"When this step started","x-format":{"isoDate":true}},"endTime":{"type":"string","description":"When this step stopped running","x-format":{"isoDate":true}}},"type":"object"}}}
```
Comment (stjohnjohnson) [inline swagger.json:1]: Any chance we could download this on-demand?

File: wercker.yml [hunk 2]
```diff
@@ -56,6 +56,12 @@ build:
           env GOOS=darwin GOOARCH=amd64 go build -a -o sd-mac
           mv sd-mac $ORG_SOURCE
 
+    - script:
+        name: go build windows
+        code: |
+          env GOOS=windows GOARCH=386 go build -a -o sd-windows
+          mv sd-windows $ORG_SOURCE
```
Comment (stjohnjohnson) [inline wercker.yml:63]: In the future, we should name these files after the system type/os (darwin, linux, etc)

**Comment by stjohnjohnson** [comment]
LGTM otherwise 👍

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me. I didn't merge it in respect of @stjohnjohnson's unanswered question

9qho8bh - imgur

File: swagger.json (hunk 1 above)
Comment (nicolaifsf) [reply swagger.json:1]: Yep we definitely can. It looks like hapi-swagger had a new version released recently. I'll check to make sure that the strings maxLength and minLength options are being generated properly. Thanks StJohn!

**Pull request merged**